relay-replays/src/recording/protocol.rs

```diff
@@ -0,0 +1,171 @@
+/// Protocol.
+///
+/// The following section includes the behaviors responsible for deserializing a recording from
+/// its HTTP transport schema and serializing a recording to its Kafka transport schema.
+///
+/// We expect recordings to come in the format of plaintext headers (JSON encoded), then a new
+/// line character, then an optionally compressed RRWeb recording.  Failure to include any
+/// component of this schema results in an error.
+use flate2::{read::ZlibDecoder, write::ZlibEncoder, Compression};
+use serde::Deserialize;
+use serde_json::Error as SError;
+use std::fmt::Display;
+use std::io::{Error, Read, Write};
+
+/// Deserializes input bytes and returns a tuple of serialized (confirmed valid) JSON headers
+/// and the optionally decompressed body value.
+pub fn deserialize(bytes: &[u8], limit: usize) -> Result<(&[u8], Vec<u8>), ProtocolError> {
```

Comment:
_nit_: This function does not actually deserialize, can we give it a better name?